File: build_pages.js
const fs = require('fs');
const cheerio = require('cheerio');

// Load index.html to grab the exact navbar
const indexHtml = fs.readFileSync('index.html', 'utf-8');
const $index = cheerio.load(indexHtml);
const newNavbar = $index('div.navbar-no-shadow-2').parent().html();

const pages = ['about', 'contact', 'imprint', 'privacy-policy', 'disclaimer-page'];

pages.forEach(page => {
    const srcFile = `src_${page}.html`;
    if (!fs.existsSync(srcFile)) return;
    
    console.log(`Processing ${srcFile}...`);
    const html = fs.readFileSync(srcFile, 'utf-8');
    const $ = cheerio.load(html);
    
    // Replace old navbar
    $('div.navbar-no-shadow-2').replaceWith(newNavbar);
    
    // Remove the language redirect scripts
    $('script').each((i, el) => {
        const text = $(el).html();
        if (text && text.includes("window.location.href = '/hr'")) {
            $(el).remove();
        }
    });

    // Remove pre-existing i18n script if any to avoid duplicates
    $('script[src="js/i18n.js"]').remove();
    
    // Inject i18n
    $('body').append('<script src="js/i18n.js"></script>\n');
    
    fs.writeFileSync(`${page}.html`, $.html());
    console.log(`Successfully built ${page}.html`);
});

File: build_pages.py
from bs4 import BeautifulSoup
import os

def build_pages():
    # Read index.html to get the navbar
    with open('index.html', 'r', encoding='utf-8') as f:
        index_soup = BeautifulSoup(f.read(), 'html.parser')
    
    navbar = index_soup.find('div', class_='navbar-no-shadow-2')
    
    pages = ['families', 'business', 'institutions', 'providers']
    
    for page in pages:
        src_file = f'src_{page}.html'
        if not os.path.exists(src_file):
            print(f"Skipping {src_file}")
            continue
            
        with open(src_file, 'r', encoding='utf-8') as f:
            soup = BeautifulSoup(f.read(), 'html.parser')
            
        # Replace navbar
        old_navbar = soup.find('div', class_='navbar-no-shadow-2')
        if old_navbar and navbar:
            old_navbar.replace_with(navbar)
            
        # Remove language redirect script
        for script in soup.find_all('script'):
            if script.string and 'window.location.href = \'/hr\'' in script.string:
                script.decompose()
                
        # Inject i18n.js
        body = soup.find('body')
        if body:
            new_script = soup.new_tag('script', src='js/i18n.js')
            body.append(new_script)
            
        # Save new html
        out_file = f'{page}.html'
        with open(out_file, 'w', encoding='utf-8') as f:
            f.write(str(soup))
        print(f"Built {out_file}")

if __name__ == '__main__':
    build_pages()
